RESTFulSense/Services/Orchestrations/Forms/FormOrchestrationService.cs

```diff
@@ -74,12 +76,62 @@ private void AddStringContents(FormModel formModel)
 
                 if (attribute != null)
                 {
-                    string value = (string)this.valueService.RetrievePropertyValue(formModel.Object, property);
-                    this.formService.AddStringContent(formModel.MultipartFormDataContent, value, attribute.Name);
+                    bool ignoreDefaultValues = attribute.IgnoreDefaultValues;
+                    object propertyValue = this.valueService.RetrievePropertyValue(formModel.Object, property);
+                    bool isDefaultValue = IsDefaultValue(property.PropertyType, propertyValue);
+
+                    if (isDefaultValue is false || ignoreDefaultValues is false)
+                    {
+                        string value = ConvertToString(propertyValue, attribute.MediaType, ignoreDefaultValues);
+                        this.formService.AddStringContent(formModel.MultipartFormDataContent, value, attribute.Name);
+                    }
                 }
             }
         }
 
+        private static bool IsDefaultValue(Type type, object value)
+        {
+            if (type.IsValueType)
+            {
+                object defaultValue = Activator.CreateInstance(type);
```

Comment:
Value types may be null, if that happens when checking equality we'll encounter a null exception. We should verify if the underlying type is Nullabe<T> and if it is then its default value is null as well.

```suggestion
                Type underlyingType = Nullable.GetUnderlyingType(type);
                bool isNullableType = underlyingType != null;
                if (isNullableType && value is null)
                {
                    return true;
                }

                object defaultValue = Activator.CreateInstance(type);
```